File: tests/inputs/hello.dart
// Sample Dart file to exercise syntax highlighting, indents,
// and textobjects in vorto. Open with `vorto assets/samples/hello.dart`.

/// Tiny program exercising typical Dart constructs:
/// classes, mixins, named/optional params, generics, futures,
/// pattern matching, records, and collection-if/for.
library;

mixin Prefixed {
  String prefix = 'Hello';
}

abstract interface class Greeter<T> {
  String greet(T value);
}

class Person {
  final String name;
  int age;
  final List<String> tags;

  Person({required this.name, this.age = 0, this.tags = const []});

  bool get isAdult => age >= 18;

  @override
  String toString() => 'Person($name, $age)';
}

class PersonGreeter with Prefixed implements Greeter<Person> {
  PersonGreeter({String? prefix}) {
    if (prefix != null) this.prefix = prefix;
  }

  @override
  String greet(Person p) => '$prefix, ${p.name}!';
}

String classify(int n) => switch (n) {
      < 0 => 'negative',
      0 => 'zero',
      _ when n.isEven => 'positive even',
      _ => 'positive odd',
    };

Future<void> main() async {
  final people = <Person>[
    Person(name: 'Alice', age: 30, tags: ['admin']),
    Person(name: 'Bob', age: 17),
    Person(name: 'Carol', age: 42, tags: ['vip', 'early-bird']),
  ];

  final greeter = PersonGreeter(prefix: 'Hi');
  for (final p in people.where((p) => p.isAdult)) {
    print(greeter.greet(p));
  }

  final adultNames = [
    for (final p in people)
      if (p.isAdult) p.name,
  ];
  print('Adults: ${adultNames.join(', ')}');

  for (var n = -2; n <= 5; n++) {
    print('$n is ${classify(n)}');
  }

  final (sum, count) = await _sumAndCount([1, 2, 3, 4]);
  print('sum=$sum count=$count');
}

Future<(int sum, int count)> _sumAndCount(Iterable<int> xs) async {
  await Future<void>.delayed(Duration.zero);
  return (xs.fold(0, (a, b) => a + b), xs.length);
}

File: tests/inputs/hello.cs
// Sample C# file to exercise syntax highlighting, indents,
// and textobjects in vorto. Open with `vorto assets/samples/hello.cs`.

using System;
using System.Collections.Generic;
using System.Linq;

namespace VortoSample;

/// <summary>
/// A tiny program that exercises typical C# constructs:
/// records, properties, pattern matching, LINQ, generics, and async.
/// </summary>
public record Person(string Name, int Age)
{
    public string Greeting => $"Hello, {Name}!";
}

public interface IGreeter<T>
{
    string Greet(T value);
}

public class PersonGreeter : IGreeter<Person>
{
    private const string DefaultPrefix = "Hello";

    public string Prefix { get; init; } = DefaultPrefix;

    public string Greet(Person p) => $"{Prefix}, {p.Name}!";
}

public static class Classifier
{
    public static string Classify(int n) => n switch
    {
        < 0 => "negative",
        0 => "zero",
        var x when x % 2 == 0 => "positive even",
        _ => "positive odd",
    };
}

public static class Program
{
    public static async Task Main()
    {
        var people = new List<Person>
        {
            new("Alice", 30),
            new("Bob", 25),
            new("Carol", 42),
        };

        var greeter = new PersonGreeter { Prefix = "Hi" };
        foreach (var p in people)
        {
            Console.WriteLine(greeter.Greet(p));
        }

        var adults = people
            .Where(p => p.Age >= 21)
            .OrderByDescending(p => p.Age)
            .Select(p => p.Name)
            .ToList();

        Console.WriteLine($"Adults: {string.Join(", ", adults)}");

        for (var n = -2; n <= 5; n++)
        {
            Console.WriteLine($"{n} is {Classifier.Classify(n)}");
        }

        await Task.Delay(0);
    }
}
